src/compass/s1_geocode_slc.py

```diff
@@ -41,6 +42,7 @@ def run(cfg):
     iters = cfg.geo2rdr_params.numiter
     blocksize = cfg.geo2rdr_params.lines_per_block
     flatten = cfg.geocoding_params.flatten
+    output_format = cfg.geocoding_params.output_format
```

Comment:
> Is validate_cloud_optimized_geotiff.py is from gdal repo? If so, do we need some license header at the start?
> 
> What do you think about the following restructure:
> 
> * move the contents of validate_cloud_optimized_geotiff into cloud_optimized_geotiff
> * rename `save_as_cog` to `save_as`
> * rename moved `validate.main` to `validate`
> * make ancillary validation functions private with `_` prefix
> * usage becomes
> 
> ```python
> import cloud_optomized_geotiff
> ...
> cloud_optomized_geotiff.save_as(...)
> ...
> cloud_optomized_geotiff.validate(...)
> ```
> 
> Depending the presence (or lack thereof of `__main__`), the suggestion above would either:
> 
> * take away ability to validate COG directly from terminal i.e. no `__main__`. (My preference for simplicity)
> * add argument parsing code `__main__` so `save_as` and `validate` can be used.

Hi @LiangJYu , yes. I was working on an older version of that script that didn't allow for API access. Apparently, we may be able to import this functionality from GDAL. See the example in:

```
import validate_cloud_optimized_geotiff.py
validate_cloud_optimized_geotiff.validate("test.tif")
```

Source: https://www.cogeo.org/developers-guide.html

I'm not sure if that import will work directly.  The source code can be found here:

https://github.com/OSGeo/gdal/blob/master/swig/python/gdal-utils/osgeo_utils/samples/validate_cloud_optimized_geotiff.py

If we have to keep this copy of the code (script), I'd suggest not changing too much its structure, since we may want to import the updates from that code from time to time.

Reply:
I don't know why GitHub moved my response in here =/

Reply:
@LiangJYu I like your suggestion, it makes sense and I will implement it. @gshiroma I was not able to import directly the script your recommended. I will follow Liang's suggestion and make sure to include a header in the validation script (sorry, I was not aware that this was taken from the GDAL repo).

Reply:
[gdal source code](https://github.com/OSGeo/gdal/blob/master/swig/python/gdal-utils/osgeo_utils/samples/validate_cloud_optimized_geotiff.py) for reference. I found it thru [here](https://www.cogeo.org/developers-guide.html) under "Testing your Cloud Optimized GeoTIFF's".